### ISSUE TITLE
Feature request: Capture output from all examples and test vs saved output

### ISSUE THREAD

**Issue by vnijs** [hadley/testthat#255]
I have a lot of functions that generate plots and print summaries. Not may testthat tests yet but lots of examples. Would be great to capture all the output from run_examples and compare to a saved run. Warnings wouldn't need to be super detailed. Just, which functions produced different output. Is this feasible?

I just noticed test_examples, which is great, but I'd like to compare the output to a known set of results.

**Comment by krlmlr** [comment]
I think this is already done by `R CMD check`, see point 18 of http://cran.r-project.org/doc/manuals/r-devel/R-exts.html#Checking-packages.

**Comment by jimhester** [comment]
@vnijs You should look at ggplot2's `visual_tests` to test the plots which uses https://github.com/wch/vtest.

[tools/R/check.R#L2577-L2589](https://github.com/wch/r-source/blob/92b24a947bef34515c955ef1a73fb61dfecade67/src/library/tools/R/check.R#L2577-L2589) has the relevant code used by `R CMD check`, looks like the function used is `tools::Rdiff()`.

**Comment by vnijs** [comment]
Thanks for the pointer @krlmlr. I figured out how to create the `-Ex.Rout.Save` file (see script below) and it is used by R CMD check as suggested. I haven't followed up on @jimhester's suggestion however. Will do that next. I noticed the `-Ex.Rout.Save` file is not used in testthat::test_examples and it seems this might be a nice feature. 

`## based on http://r.789695.n4.nabble.com/timings-for-examples-in-R-CMD-check-td4696443.html
cd ~/gh/
rm -f radiant/tests/Examples/radiant-Ex.Rout.save
R CMD build radiant
R CMD check radiant*.tar.gz
mv radiant.Rcheck/radiant-Ex.Rout radiant/tests/Examples/radiant-Ex.Rout.save
rm -rf radiant.Rcheck
cd ~/gh/radiant`
